paddle/cinn/hlir/dialect/operator/transforms/add_broadcast_to_elementwise_pass.cc

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/cinn/hlir/dialect/operator/transforms/add_broadcast_to_elementwise_pass.h"
+
+#include "paddle/cinn/hlir/dialect/operator/ir/cinn_op.h"
+#include "paddle/fluid/pir/dialect/operator/ir/pd_op.h"
+#include "paddle/fluid/pir/drr/api/drr_pattern_base.h"
+#include "paddle/fluid/pir/drr/api/match_context.h"
+#include "paddle/phi/core/ddim.h"
+#include "paddle/pir/core/builtin_dialect.h"
+#include "paddle/pir/pass/pass.h"
+#include "paddle/pir/pass/pass_manager.h"
+#include "paddle/pir/pattern_rewrite/pattern_applicator.h"
+#include "paddle/pir/pattern_rewrite/pattern_match.h"
+#include "paddle/pir/pattern_rewrite/pattern_rewrite_driver.h"
+
+namespace cinn {
+namespace dialect {
+namespace ir {
+
+int64_t GetDimByIndex(const phi::DDim& first,
+                      const phi::DDim& second,
+                      int short_align_axis,
+                      int idx) {
+  // rank of first less than rank of second
+  if (idx < short_align_axis) {
+    return second[idx];
+  } else {
+    return first[idx - short_align_axis] > second[idx]
+               ? first[idx - short_align_axis]
+               : second[idx];
+  }
+}
+
+std::vector<int64_t> GetOutputShape(const phi::DDim& x, const phi::DDim& y) {
+  std::vector<int64_t> vec_res;
+  if (x.size() >= y.size()) {
+    int short_align_axis = x.size() - y.size();
+    int max_rank = x.size();
+    vec_res.resize(max_rank);
+    for (size_t i = 0; i < max_rank; ++i) {
+      vec_res[i] = GetDimByIndex(y, x, short_align_axis, i);
+    }
+  } else {
+    int short_align_axis = y.size() - x.size();
+    int max_rank = y.size();
+
+    vec_res.resize(max_rank);
+    for (size_t i = 0; i < max_rank; ++i) {
+      vec_res[i] = GetDimByIndex(x, y, short_align_axis, max_rank);
+    }
+  }
+
+  return vec_res;
+}
+
+bool is_same_dim(const phi::DDim& first, const std::vector<int64_t>& second) {
```

Comment:
```suggestion
bool IsSameDim(const phi::DDim& first, const std::vector<int64_t>& second) {
```
前面的函数是「大写+驼峰」的，这里可以保持一致？

Reply:
done